### PR TITLE
CA-388398: cores-per-socket causes a massive performance hit, remove it

### PR DESCRIPTION
**Pull request by edwintorok**
This got introduced in 6822644, to ensure that VMs have a single socket by default. However that'd be the default *without* a cores-per-socket setting anyway (1 socket, all vcpus as cores).

In CA-388398 we've measured an ~60% performance difference in an Apache Siege benchmark with all cores-per-socket settings, except:
* cores-per-socket==vcpu count
* cores-per-socket absent

Although that measurement was done on Linux, the setting is expected to have a detrimental effect elsewhere, the topology changes that Xen has to make in order to implement cores-per-socket results in a topology that is not quite correct.

Although cores-per-socket==vcpu with the setting here, as soon as the user adds more vCPUs that won't be true anymore.

Fixes: 6822644 ("Update base Windows templates to 2 vCPUs")

### CODE REVIEW

**Comment by edwintorok** [comment]
Draft, needs to be tested

**Comment by edwintorok** [comment]
Looks like there is a xencenter bug, where if cores-per-socket is not specified it defaults to N sockets with 1 core (the worst possible one for performance). Whereas XAPI/Xen would default to N cores, 1 socket.

**Comment by edwintorok** [comment]
We'll need to wait for a XenCenter bugfix before we can merge this

**Comment by edwintorok** [comment]
actually due to a different bug it is "fine": xencenter shows it will use 1 core per socket, but if you leave that dropdown unchanged it won't write a value into the platform field, which means that Xen will use the optimal N cores per socket.
Lets wait until the bug is fixed though otherwise this'd just be very confusing to users.

**Review by rosslagerwall — APPROVED** [review]

**Comment by edwintorok** [comment]
There are too many bugs in this area, will reopen once they are fixed in xencenter/xen

**Pull request closed without merging**
